packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/undo/services/recorder/EditRectangularNodeAppearanceRecorder.java

```diff
@@ -0,0 +1,121 @@
+import org.eclipse.sirius.components.diagrams.events.appearance.NodeBorderColorAppearanceChange;
+import org.eclipse.sirius.components.diagrams.events.appearance.NodeBorderRadiusAppearanceChange;
+import org.eclipse.sirius.components.diagrams.events.appearance.NodeBorderSizeAppearanceChange;
+import org.eclipse.sirius.components.diagrams.events.appearance.NodeBorderStyleAppearanceChange;
+import org.eclipse.sirius.components.diagrams.events.appearance.ResetNodeAppearanceChange;
+import org.eclipse.sirius.components.representations.IRepresentation;
+import org.eclipse.sirius.web.application.editingcontext.EditingContext;
+import org.eclipse.sirius.web.application.services.api.IDiagramGraphicalChangesRecorderHandler;
+import org.eclipse.sirius.web.application.undo.events.DiagramNodeAppearanceEventRecord;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Use to record DiagramNodeAppearanceEventRecord from EditRectangularNodeAppearanceInput.
+ *
+ * @author mcharfadi
+ */
+@Service
+public class EditRectangularNodeAppearanceRecorder implements IDiagramGraphicalChangesRecorderHandler {
+
+    private final IDiagramQueryService diagramQueryService;
+
+    public EditRectangularNodeAppearanceRecorder(IDiagramQueryService diagramQueryService) {
+        this.diagramQueryService = diagramQueryService;
+    }
+
+    @Override
+    public void recordChange(IEditingContext editingContext, IRepresentation previousRepresentation, IDiagramEvent diagramEvent, ChangeDescription changeDescription) {
+        if (diagramEvent instanceof EditAppearanceEvent editAppearanceEvent && changeDescription.getInput() instanceof IDiagramInput diagramInput) {
+            List<IAppearanceChange> undoAppearanceChanges = new ArrayList<>();
+            List<IAppearanceChange> redoAppearanceChanges = new ArrayList<>(editAppearanceEvent.changes());
+
+            if (editingContext instanceof EditingContext siriusEditingContext && previousRepresentation instanceof Diagram previousDiagram) {
+                editAppearanceEvent.changes().stream().filter(INodeAppearanceChange.class::isInstance)
+                    .map(INodeAppearanceChange.class::cast)
+                    .forEach(change -> this.handleAppearanceChange(undoAppearanceChanges, previousDiagram, change));
+
+                if (!undoAppearanceChanges.isEmpty()) {
+                    var nodeChange = new DiagramNodeAppearanceEventRecord(diagramInput.id().toString(), diagramInput.representationId(), undoAppearanceChanges, redoAppearanceChanges);
+                    siriusEditingContext.getInputId2GraphicalChange().put(diagramInput.id().toString(), nodeChange);
+                }
+            }
+        }
+    }
+
+    private void handleAppearanceChange(List<IAppearanceChange> undoAppearanceChanges, Diagram previousDiagram, INodeAppearanceChange change) {
```

Comment:
There's no need to modify an argument like that, just return a proper result instead